client/state/data-layer/wpcom/sites/rewind/credentials/index.js

```diff
@@ -0,0 +1,53 @@
+import { translate } from 'i18n-calypso';
+import {
+	JETPACK_CREDENTIALS_TEST,
+	JETPACK_CREDENTIALS_TEST_FAILURE,
+	JETPACK_CREDENTIALS_TEST_SUCCESS,
+} from 'calypso/state/action-types';
+import { registerHandlers } from 'calypso/state/data-layer/handler-registry';
+import { http } from 'calypso/state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'calypso/state/data-layer/wpcom-http/utils';
+import { errorNotice } from 'calypso/state/notices/actions';
+
+export const testCredentials = ( action ) =>
+	http(
+		{
+			apiNamespace: 'wpcom/v2',
+			method: 'POST',
+			path: `/sites/${ action.siteId }/rewind/credentials/test`,
+			query: {
+				role: action.role,
+			},
+			body: {},
+		},
+		action
+	);
+
+export const testCredentialsSucceeded = ( { siteId, role }, { ok } ) => ( {
+	type: JETPACK_CREDENTIALS_TEST_SUCCESS,
```

Comment:
I'd suggest making sure that any actions that are dispatched anywhere (including by the data layer) are created by action creator functions, declared in `actions.js`. This allows a single update path towards debugging and editing a particular action, and that reduces the ambiguity and the potential for errors.

Reply:
So the idea is to declare those functions on `actions.js` and call them from anywhere using something like `dispatch( testCredentials( siteId, role ) );` or `dispatch( markCredentialsAsInvalid( siteId, role ) );`, am I right?

I will be on it.

Reply:
Yes 👍